py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py

```diff
@@ -1759,6 +1759,30 @@ def aten_ops_logical_not(
     )
 
 
+# TODO: confirm with Dheeraj, whether it requires supports_dynamic_shapes=True
+# according to https://pytorch.org/docs/stable/generated/torch.sym_not.html
+# Boolean value of Tensor with more than one value is not allowed, so it must be static_shaped
+# eg. allowed: True or torch.tensor(True) or torch.tensor([False]) or torch.tensor([[False]])
+# eg. not allowed: torch.tensor([True, False]) or torch.tensor([[True], [False]])
```

Comment:
You can enable supports_dynamic_shapes=True in this case so that there's no segmentation. Most likely, there wouldn't be a scenario where input is a dynamic list for this op (your code throws an error here anyways, so that's fine).